client/src/main/java/io/pravega/client/segment/impl/SegmentOutputStreamImpl.java

```diff
@@ -390,7 +395,7 @@ public void write(PendingEvent event) {
             connection.send(new Append(segmentName, writerId, eventNumber, Unpooled.wrappedBuffer(event.getData()),
                                        event.getExpectedOffset()));
         } catch (ConnectionFailedException e) {
-            log.warn("Connection failed due to: ", e);
+            log.warn("Connection " + writerId + " failed due to: ", e);
```

Comment:
use `{}` instead of concatenation; it will only get invoked if the log is enabled vs every time